File: apps/client/lib/features/auth/application/usecases/register_usecase.dart
import '../../domain/entities/auth_session.dart';
import '../../domain/repositories/auth_repository.dart';

class RegisterUseCase {
  const RegisterUseCase(this._repo);
  final AuthRepository _repo;
  Future<AuthSession> call({
    required String fullName,
    required String email,
    required String password,
  }) =>
      _repo.register(fullName: fullName, email: email, password: password);
}

File: apps/backend/src/modules/auth/application/usecases/register.usecase.ts
import { Injectable } from '@nestjs/common';
import { AuthRepository } from '../../domain/repositories/auth.repository';
import { AuthSession } from '../../domain/entities/auth-session.entity';

@Injectable()
export class RegisterUseCase {
  constructor(private readonly repo: AuthRepository) {}
  execute(input: { fullName: string; email: string; password: string }): Promise<AuthSession> {
    return this.repo.register(input);
  }
}
